pkg/apis/v1alpha1/nsxserviceaccount_types.go

```diff
@@ -0,0 +1,86 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NSXServiceAccountSpec defines the desired state of NSXServiceAccount
+type NSXServiceAccountSpec struct {
+	VPCName string `json:"vpcName,omitempty"`
+}
+
+type NSXProxyEndpointAddress struct {
+	Hostname string `json:"hostname,omitempty"`
+	//+kubebuilder:validation:Format=ip
+	IP string `json:"ip,omitempty"`
+}
+
+type NSXProxyProtocol string
+
+const (
+	NSXProxyProtocolTCP NSXProxyProtocol = "TCP"
+)
+
+type NSXProxyEndpointPort struct {
+	Name     string           `json:"name,omitempty"`
+	Port     uint16           `json:"port,omitempty"`
+	Protocol NSXProxyProtocol `json:"protocol,omitempty"`
```

Comment:
This, or the type `NSXProxyProtocol`, should include an enum validator, ex.

```go
// +kubebuilder:validation:Enum=TCP
type NSXProxyProtocol string
```

or

```go
// +kubebuilder:validation:Enum=TCP
Protocol NSXProxyProtocol `json:"protocol,omitempty"`
```

Generally it's a good idea to place the enum validator on the type so the validation occurs wherever the field is used.